### ISSUE TITLE
Duplicate Pipeline Errors on Config repo SPA are not removed when the existing config xml pipeline is deleted

### ISSUE THREAD

**Issue by adityasood** [gocd/gocd#5759]
##### Issue Type
- Bug Report

##### Summary

The Config merge for config repo and cruise-config.xml produces duplicate pipeline error if the pipeline name is duplicate across both config repo and cruise-config.

The error on the config repo SPA does not go away if the duplicate pipeline in cruise-config.xml is deleted. The error is removed once the user makes another commit on the said config repo

###### Basic environment details

* Go Version:  19.1.0-8451
* JAVA Version:Java10
* OS: MacOS
* Browser vendor and version (if relevant): Chrome

##### Steps to Reproduce

1. Create pipeline xml-pipeline-23 using the UI.
2. Use the export API to export this pipeline in YAML/JSON format
3. Commit this exported pipeline into a git repo
4. Add this git repo as a config repo in GoCD
5. Wait for GoCD to parse the config-repo
6. You should now see parse errors
7. Delete the duplicate pipeline from UI
8. You will still see parse errors on the repo
9. Make a dummy commit to the git repo and you will see the error gone once the parse is done

**Comment by arvindsv** [comment]
Closing as a duplicate of #5792.
